inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java

```diff
@@ -166,6 +169,21 @@ public Map<String, String> tableOptions() {
         return options;
     }
 
+    /**
+     * true is upsert kafka connector
+     * false is kafka connector
+     * @return connector option is
+     */
+    private boolean isUpsertKafkaConnector(Format format, boolean hasPrimaryKey) {
+        if (format instanceof JsonFormat && hasPrimaryKey) {
```

Comment:
Maybe it is better implement by 'return hasPrimaryKey && (format instansof JsonFormat || format instansof ...);'

Reply:
> Maybe it is better implement by 'return hasPrimaryKey && (format instansof JsonFormat || format instansof ...);'

It is too long. It could add other format in th future.